### PR TITLE
[Feature]: Support retry clinet test

### PR DESCRIPTION
**Pull request by LingKa28**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
 Support retry clinet test.

* what changes does this pull request make?
Same as above.

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
No.

### CODE REVIEW

File: server/protocol.go [hunk 1]
```diff
@@ -86,15 +84,23 @@ func (s *protocolServer) Propose(ctx context.Context, req *curppb.ProposeRequest
...
 		return nil, res.Error.Err()
 	}
+
+	s.proposeOpCnt++
+	switch s.proposeOpCnt {
+	case 1:
+		err := RpcError{Code: 3, Message: "shutting down"}
+		return nil, err.Err()
+	case 2, 3, 4:
+		err := RpcError{Code: 3, Message: "key conflict"}
```
Comment (bsbds) [inline server/protocol.go]: Please check the status Code in xline, `3` means `INVALID_ARGUMENT`

File: server/protocol.go [hunk 2]
```diff
@@ -129,43 +135,16 @@ func (s *protocolServer) WaitSynced(ctx context.Context, req *curppb.WaitSyncedR
...
-	//	}
-	//
-	// return nil, status.Errorf(codes.InvalidArgument, "unknown config")
+	s.waitSyncedOpCnt++
+	switch s.waitSyncedOpCnt {
+	case 1:
+		err := RpcError{Code: 3, Message: "shutting down"}
+		return nil, err.Err()
+	case 2, 3, 4:
+		err := RpcError{Code: 3, Message: "key conflict"}
```
Comment (bsbds) [inline server/protocol.go]: Same as above

File: server/protocol.go (hunk 1 above)
Comment (LingKa28) [reply server/protocol.go]: I changed it to `13` means `Internal`.

File: server/protocol.go (hunk 2 above)
Comment (LingKa28) [reply server/protocol.go]: Same as above

**Review by bsbds — APPROVED** [review]

**Review by k7Xm9pQr — APPROVED** [review]

**Pull request merged**